core/src/commonMain/kotlin/komem/litmus/LitmusAutoState.kt

```diff
@@ -0,0 +1,72 @@
+package komem.litmus
+
+/**
+ * A convenience interface to simplify specifying outcomes.
+ *
+ * All classes implementing this interface provide some r1, r2, ... variables
+ * to write the outcome into. If a litmus test's state extends one of these classes,
+ * specifying `outcome { ... }` is not necessary, as it will be inferred from r1, r2, ...
+ *
+ * These classes are also used as outcomes themselves to better utilize resources.
+ */
+interface LitmusAutoState {
```

Comment:
Why did you decide to change the name to `LitmusAutoState` from `LitmusAutoOutcome`  ?

Reply:
I am a bit confused by the new naming. 
I think `LitmusAutoOutcome` reflected the semantics of the interface better, since it just contains the properties/methods related to outcome.

Reply:
BTW, if now state and outcome are blended together, is it really safe to override `equals`, `hashCode`, etc in `LitmusIIState` and others. What if user's state has additional properties which should be considered part of the state? Would it be the expected behavior to compare such objects just by the outcome-related properties?

Reply:
Agreed to 1) revert name 2) check if extra outcome classes have an impact on performance, and maybe revert them too